mlflow/exceptions.py

```diff
@@ -4,3 +4,28 @@ class MlflowException(Exception):
 
 class IllegalArtifactPathError(MlflowException):
     """The artifact_path parameter was invalid."""
+
+
+class ExecutionException(MlflowException):
+    """Exception thrown when executing a project fails."""
+    pass
+
+
+class DownloadException(MlflowException):
+    """Exception thrown when downloading an artifact fails."""
+    pass
+
+
+class ShellCommandException(MlflowException):
+    """Exception thrown when executing a shell command fails."""
+    pass
+
+
+class RestException(MlflowException):
```

Comment:
As discussed, maybe we could try creating `exceptions/__init__.py`, and import this from there, so we can keep the implementation of the exception next to the thing that uses it.